subsys/config/src/config.c

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include <errno.h>
+
+#include "mbedtls/base64.h"
+
+#include "config/config.h"
+#include "config_priv.h"
+#include <zephyr/types.h>
+
+/* mbedtls-base64 lib encodes data to null-terminated string */
+#define BASE64_ENCODE_SIZE(in_size) ((((((in_size) - 1) / 3) * 4) + 4) + 1)
+
+sys_slist_t conf_handlers;
+
+static u8_t conf_cmd_inited;
+
+void conf_init(void)
+{
+	sys_slist_init(&conf_handlers);
+	conf_store_init();
+
+	if (conf_cmd_inited) {
```

Comment:
should we do this check at the beginning of the function? If not then we can remove this check as it is not doing anything worthwhile.

Reply:
moved to the function's begin